sdk/tests/conformance/extensions/ext-texture-mirror-clamp-to-edge.html

```diff
@@ -0,0 +1,213 @@
+    debug("");
+    debug("Check enums");
+    shouldBe("ext.MIRROR_CLAMP_TO_EDGE_EXT", "0x8743");
+}
+
+function checkQueries() {
+    debug("");
+    debug("Check texture and sampler state updates");
+
+    const tex = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_2D, tex);
+
+    for (const pname of pnames) {
+        gl.texParameteri(gl.TEXTURE_2D, gl[pname], ext.MIRROR_CLAMP_TO_EDGE_EXT);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors from texParameteri");
+        shouldBe(`gl.getTexParameter(gl.TEXTURE_2D, gl.${pname})`, "ext.MIRROR_CLAMP_TO_EDGE_EXT");
+        gl.texParameteri(gl.TEXTURE_2D, gl[pname], gl.REPEAT);
+
+        gl.texParameterf(gl.TEXTURE_2D, gl[pname], ext.MIRROR_CLAMP_TO_EDGE_EXT);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors from texParameterf");
+        shouldBe(`gl.getTexParameter(gl.TEXTURE_2D, gl.${pname})`, "ext.MIRROR_CLAMP_TO_EDGE_EXT");
+        gl.texParameterf(gl.TEXTURE_2D, gl[pname], gl.REPEAT);
+    }
+
+    if (!gl.createSampler) return;
+
+    sampler = gl.createSampler();
+    for (const pname of pnames) {
+        gl.samplerParameteri(sampler, gl[pname], ext.MIRROR_CLAMP_TO_EDGE_EXT);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors from samplerParameteri");
+        shouldBe(`gl.getSamplerParameter(sampler, gl.${pname})`, "ext.MIRROR_CLAMP_TO_EDGE_EXT");
+        gl.samplerParameteri(sampler, gl[pname], gl.REPEAT);
+
+        gl.samplerParameterf(sampler, gl[pname], ext.MIRROR_CLAMP_TO_EDGE_EXT);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors from samplerParameterf");
+        shouldBe(`gl.getSamplerParameter(sampler, gl.${pname})`, "ext.MIRROR_CLAMP_TO_EDGE_EXT");
+        gl.samplerParameterf(sampler, gl[pname], gl.REPEAT);
+    }
+}
+
+function checkSampling() {
+    debug("");
+    debug(`Check texture sampling with mirror-clamp-to-edge mode`);
+
+    wtu.setupUnitQuad(gl);
+    const vs = `precision highp float;
+        attribute vec4 vPosition;
+        varying vec2 texCoord;
+        void main() {
+            gl_Position = vec4(vPosition.xy, 0.0, 1.0);
+            texCoord = vPosition.xy * 2.0;
+        }`;
+    const program = wtu.setupProgram(gl, [vs, wtu.simpleTextureFragmentShader]);
+    gl.useProgram(program);
+
+    const black = [  0,   0,   0, 255];
+    const red   = [255,   0,   0, 255];
+    const green = [  0, 255,   0, 255];
+    const blue  = [  0,   0, 255, 255];
+    const data = new Uint8Array([...black, ...red, ...green, ...blue]);
+
+    function checkPixels() {
+        function checkPixel(x, y, color) {
+            const screen = (s, t) => s * (t * 0.5 + 0.5);
+            wtu.checkCanvasRect(gl, screen(w, x), screen(h, y), 1, 1, color,
+                                `(${x.toFixed(3)}, ${y.toFixed(3)}): ${color} `);
+        }
+        for (const signX of [+1, -1]) {
+            for (const signY of [+1, -1]) {
+                checkPixel(signX * 0.125, signY * 0.125, black);
```

Comment:
The meaning of these coefficients is probably obvious when running the tests, but they're confusing when reviewing the code. Comments would help.

Reply:
Done.

Reply:
Thanks.